ramips: switch default SDHC driver to upstream version

DragonBluep opened the pull request.

neheb commented on target/linux/ramips/patches-6.6/831-mmc-mtk-sd-initialize-pad-delay-and-drive-strength.patch:
```diff
@@ -0,0 +1,33 @@
++	/* Set pad delay */
++	sdr_clr_bits(host->base + MSDC_PAD_TUNE, 0xffdf1f1f);
++	sdr_set_bits(host->base + MSDC_PAD_TUNE, 0x84101010);
++	writel(0x10101010, host->base + MSDC_PAD_TUNE0);
```
isn't tune0 handled already?

DragonBluep replied:
MT762x only operates on register `MSDC_PAD_TUNE`. ref:
https://github.com/torvalds/linux/blob/256abd8e550ce977b728be79a74e1729438b4948/drivers/mmc/host/mtk-sd.c#L539

In our local driver. This register name `MSDC_DAT_RDDLY0`.
https://github.com/openwrt/openwrt/blob/e7599942af8e14541166234349e9e49278455ef6/target/linux/ramips/files/drivers/mmc/host/mtk-mmc/sd.c#L2107

mrkiko commented:
Nice work.

DragonBluep committed:
ramips: make SoC dtsi compatible with upstream mmc-mtk driver

Add all essential MTK SDHC properties to support the new mmc-mtk
driver. Since this driver relies on power regulators, we also
need to enable this feature for MT7620, just like MT762{1,8}.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Co-authored-by: John Thomson <git@johnthomson.fastmail.com.au>

DragonBluep committed:
ramips: modules: add Mediatek MMC driver package

This is the upstream implementation of the MTK SD/SDIO/MMC card
reader driver. It is an alternative solution for the downstream
driver package "kmod-sdhci-mt7620".

Signed-off-by: Shiji Yang <yangshiji66@qq.com>
Co-authored-by: John Thomson <git@johnthomson.fastmail.com.au>

DragonBluep committed:
ramips: mtk-mmc: use "cd-inverted" to set CD line polarity

"cd-inverted" is an upstream documented property used to indicate
the CD line is actived high. We will introduce a new upstream SDHC
driver, and this change will make them compatible with each other.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ramips: switch default SDHC driver to kmod-mmc-mtk

Now that the SDHC of MT762{0,1,8} has been supported upstream, it's
time to switch the default driver to the upstream one. We will still
keep the old driver for users to choose from.

Tested on HiWiFi HC5861 (MT7620A) and HiWiFi HC5661A (MT7628N).

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ramips: mtk-sd: initialize pad delay and drive strength

Port vendor register init values to upstream MTK SHDC driver.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ramips: improve support for HiWiFi HC5861 SDHC

1. Override max clock frequency to a stable value 24 MHz.
2. Use voltage regulator to control the power supply.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep commented:
@john-tho @PolynomialDivision Based on https://github.com/openwrt/openwrt/pull/12738#discussion_r1278314904, I made some improvements. After testing, now it works well on MT7620 and MT7628.

fkroessbacher commented:
Hy out there,

i'm on an MikroTik RouterBOARD 750Gr3 (MediaTek MT7621 ver:1 eco:4, ramips/mt7621) and today i tried the new Release OpenWrt 23.05.4, but it's not working. So i Switched back to the only working solution OpenWrt 22.03.6.

Should the Fix be in the new Version, or missed i something?

cheers

DragonBluep commented:
> Should the Fix be in the new Version, or missed i something?

@fkroessbacher Only in snapshot version.

noda-odon commented:
> @fkroessbacher Only in snapshot version.

@DragonBluep , today, I downloaded snapshot Tue Aug 20 01:33:07 2024
and the issue is still there

PaddyTheShamrock commented:
> > @fkroessbacher Only in snapshot version.
> 
> @DragonBluep , today, I downloaded snapshot Tue Aug 20 01:33:07 2024 and the issue is still there

Try:

`opkg remove kmod-sdhci-mt7620`

`opkg install kmod-mmc-mtk`

Then reboot the router.

noda-odon commented:
@PaddyTheShamrock , It works!
Thanks

BTW: I have two mmc cards, one have the reported issue, while another works OK without replacing modules.
